Trabalho Final de Mineração de Dados
Resumo: Estudo de caso aplicado em uma base de dados real do setor de saúde. O objetivo é utilizar o processo de KDD (Knowledge Discovery in Databases) para a identificação de padrões, predição e extração de conhecimento sobre patologias cardíacas em crianças e jovens.
Aplicando os processos de Seleção de dados alvo, pré-processamento e limpeza, transformação, mineração de dados e Avaliação.

1° parte: analisar o que temos dentro do dataset e as variações de dados 

In [ ]:
import pandas as pd

df = pd.read_csv('UCMF_raw.csv')

# Loop para analisar cada coluna separadamente de forma isolada
for coluna in df.columns:
    print("\n" + "="*70)
    print(f"  ANÁLISE DE CADA VARIÁVEL: {coluna}")
    print("="*70)
    
    # 1. Informações básicas de tipo e preenchimento
    tipo = df[coluna].dtype
    total_linhas = len(df)
    nulos = df[coluna].isnull().sum()
    
    print(f"• Tipo de dado na coluna: {tipo}")
    print(f"• Linhas preenchidas: {df[coluna].count()} / {total_linhas}")
    print(f"• Linhas vazias (nulas): {nulos} ({ (nulos/total_linhas)*100 :.2f}%)")
    
    # 2. Extração de todos os valores diferentes (Únicos)
    # .dropna() remove os nulos temporariamente apenas para listar os valores reais existentes
    valores_distintos = df[coluna].dropna().unique()
    qtd_distintos = len(valores_distintos)
    
    print(f"• Quantidade de valores distintos/diferentes nesta coluna: {qtd_distintos}")
    print("-" * 70)
    
    # CASO A: Colunas com poucos valores diferentes (Ex: SEXO, SOPRO, B2, NORMAL X ANORMAL)
    if qtd_distintos <= 35:
        print("-> LISTA DE TODOS OS VALORES DIFERENTES PRESENTES:")
        print(valores_distintos)
        
        print("\n-> DISTRIBUIÇÃO E FREQUÊNCIA DE CADA VALOR (Incluindo Nulos):")
        frequencia = df[coluna].value_counts(dropna=False)
        percentual = df[coluna].value_counts(dropna=False, normalize=True) * 100
        
        tabela_separada = pd.DataFrame({
            'Quantidade (Qtd)': frequencia,
            'Proporção (%)': percentual.round(2)
        })
        print(tabela_separada)
        
    # CASO B: Colunas contínuas ou identificadores com muitos valores (Ex: ID, IMC, IDADE, FC)
    else:
        print(f"Nota: Como esta coluna possui muitos valores diferentes ({qtd_distintos}),")
        print("exibir todos criaria uma lista muito longa. Veja os 10 primeiros exemplos de valores únicos:")
        print(valores_distintos[:10])
        
        # Se for numérico, nos mostra as fronteiras (mínimo e máximo)
        if pd.api.types.is_numeric_dtype(df[coluna]):
            print("\n-> RESUMO DOS EXTREMOS NUMÉRICOS:")
            print(f"   - Menor valor real encontrado: {df[coluna].min()}")
            print(f"   - Maior valor real encontrado: {df[coluna].max()}")
            print(f"   - Média geral da coluna: {df[coluna].mean():.2f}")
            print(f"   - Mediana (valor do meio): {df[coluna].median():.2f}")
            
    print("\n" + "."*70) # Divisor de fim de bloco de coluna


  ANÁLISE ISOLADA DA VARIÁVEL: ID
• Tipo de dado na coluna: int64
• Linhas preenchidas: 12873 / 12873
• Linhas vazias (nulas): 0 (0.00%)
• Quantidade de valores distintos/diferentes nesta coluna: 12873
----------------------------------------------------------------------
Nota: Como esta coluna possui muitos valores diferentes (12873),
exibir todos criaria uma lista muito longa. Veja os 10 primeiros exemplos de valores únicos:
[ 1  2  3  4  7  8  9 10 11 12]

-> RESUMO DOS EXTREMOS NUMÉRICOS:
   - Menor valor real encontrado: 1
   - Maior valor real encontrado: 17873
   - Média geral da coluna: 9196.41
   - Mediana (valor do meio): 9222.00

......................................................................

  ANÁLISE ISOLADA DA VARIÁVEL: Peso
• Tipo de dado na coluna: float64
• Linhas preenchidas: 12555 / 12873
• Linhas vazias (nulas): 318 (2.47%)
• Quantidade de valores distintos/diferentes nesta coluna: 627
----------------------------------------------------------------------
N

O que encontramos após essa análise inicial?
- ID: Variável irrelevante para o processo;
- Peso: Possui dados faltantes (2.47%) e possui peso negativo, o que não é possível;
- Altura: Possui altura nula, o que não é possível;
- IMC: Possui dados faltantes (36.72%), caso tenhamos peso e altura, é possível calcular, menor valor foi 0;
- Atendimento: Pode ser relevante caso idade do paciente não foi coletado, dados faltantes (7.64%);
- Data de nascimento: Possui dados faltantes (10.69%), pode ser prejudicial pois o estudo é por faixa etária;
- Idade: Possui dados faltantes (11.62%), e com valores negativos, o que não é possível;
- Convênio: Variável irrelevante para o processo;
- Pulsos: Possui dados faltantes (9.28%), com palavras igual escritas de formas diferentes;
- PA sistólica: Possui muitos dados faltantes (60.05%), pode atrapalhar o processo;
- PA diastólica: Possui muitos dados faltantes (60.13%), pode atrapalhar o processo;
- PPA: Pode substituir os PAs, mas possui dados faltantes (1.69%), não calculado (70.54%) e sem valor (11.62%);
- Patologia: Possui dados faltantes (9.07%), e palavras iguais escritas de formas diferentes;
- B2: Possui dados faltantes (9.15%);
- Sopro: Possui dados faltantes (9.07%) e palavras iguais escritas de formas diferentes;
- Frequencia cardíaca: Possui dados faltantes (14.51%);
- Histórico de Doença 1: Possui dados faltantes (33.17%);
- Histórico de Doença 2: Possui dados faltantes (96.84%);
- Sexo: Poucos dados faltantes (0.03%), mas palavras iguais escritas de formas diferentes;
- Motivo 1: Possui dados faltantes (8.20%);
- Motivo 2: Possui dados faltantes (27.16%);